### PR TITLE
fix(schema): change `redirect` type for `NuxtPage` type

### LINKED ISSUE #21706: NuxtPage redirect type is wrong for pages extend hook
### Environment

-

### Reproduction

https://stackblitz.com/edit/github-yb5amz?file=nuxt.config.ts

### Describe the bug

When extend pages got type error for `NuxtPage` type

```Typescript
type NuxtPage = {
    name?: string;
    path: string;
    file?: string;
    meta?: Record<string, any>;
    alias?: string[] | string;
    redirect?: string;
    children?: NuxtPage[];
};
```

Must be

```Typescript
import type { RouteRecordRedirectOption } from 'vue-router'

type NuxtPage = {
    name?: string;
    path: string;
    file?: string;
    meta?: Record<string, any>;
    alias?: string[] | string;
    redirect?: RouteRecordRedirectOption; // or another redirect type
    children?: NuxtPage[];
};
```

### Additional context

Can I create a pull request for this?

or tell me how it can be fixed more correctly :)

### Logs

_No response_

### PR DESCRIPTION
**Pull request by silencerspirit**
### 🔗 Linked issue

#21706 
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Resolve correct type for `redirect` key in `NuxtPage` type.
Resolves #21706

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/schema/src/types/hooks.ts [hunk 1]
```diff
@@ -25,7 +26,7 @@ export type NuxtPage = {
   file?: string
   meta?: Record<string, any>
   alias?: string[] | string
-  redirect?: string
+  redirect?: RouteRecordRedirectOption
```
Comment (danielroe) [inline packages/schema/src/types/hooks.ts:29]:
```suggestion
  redirect?: RouteLocationRaw
```

**Review by danielroe — COMMENTED** [review]
We need to exclude functions as they can't be stringified. Either `RouteLocationRaw` or `Exclude<RouteRecordRedirectOption, Function>` would likely do the trick, though I haven't tested the latter option.

**Comment by silencerspirit** [comment]
> We need to exclude functions as they can't be stringified. Either `RouteLocationRaw` or `Exclude<RouteRecordRedirectOption, Function>` would likely do the trick, though I haven't tested the latter option.

Thx for review, i set `RouteLocationRaw` but `Exclude<RouteRecordRedirectOption, Function>` - works to)

**Review by danielroe — APPROVED** [review]

**Comment by danielroe** [comment]
Thank you ❤️

**Pull request merged**